COMETwebapp/Components/Shared/ParameterTypeEditors/QuantityKindParameterTypeEditor.razor.cs

```diff
@@ -0,0 +1,88 @@
+//     License as published by the Free Software Foundation; either
+//     version 3 of the License, or (at your option) any later version.
+// 
+//     The COMET WEB Community Edition is distributed in the hope that it will be useful,
+//     but WITHOUT ANY WARRANTY; without even the implied warranty of
+//     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//    Affero General Public License for more details.
+// 
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//  </copyright>
+//  --------------------------------------------------------------------------------------------------------------------
+
+using COMETwebapp.ViewModels.Components.Shared.ParameterEditors;
+
+namespace COMETwebapp.Components.Shared.ParameterTypeEditors
+{
+    using CDP4Common.EngineeringModelData;
+    using CDP4Common.SiteDirectoryData;
+
+    using Microsoft.AspNetCore.Components;
+
+    using ReactiveUI;
+
+    /// <summary>
+    /// Support class for the <see cref="QuantityKindParameterTypeEditor"/>
+    /// </summary>
+    public partial class QuantityKindParameterTypeEditor
+    {
+        /// <summary>
+        /// Gets or sets the <see cref="IParameterEditorBaseViewModel{T}"/>
+        /// </summary>
+        [Parameter]
+        public IParameterEditorBaseViewModel<QuantityKind> ViewModel { get; set; }
+
+        /// <summary>
+        /// Event Callback for when a value has changed on the parameter
+        /// </summary>
+        [Parameter]
+        public EventCallback<IValueSet> ParameterValueChanged { get; set; }
+
+        /// <summary>
+        /// Method invoked after each time the component has been rendered. Note that the component does
+        /// not automatically re-render after the completion of any returned <see cref="Task"/>, because
+        /// that would cause an infinite render loop.
+        /// </summary>
+        /// <param name="firstRender">
+        /// Set to <c>true</c> if this is the first time <see cref="OnAfterRender(bool)"/> has been invoked
+        /// on this component instance; otherwise <c>false</c>.
+        /// </param>
+        /// <returns>A <see cref="Task"/> representing any asynchronous operation.</returns>
+        /// <remarks>
+        /// The <see cref="OnAfterRender(bool)"/> and <see cref="OnAfterRenderAsync(bool)"/> lifecycle methods
+        /// are useful for performing interop, or interacting with values received from <c>@ref</c>.
+        /// Use the <paramref name="firstRender"/> parameter to ensure that initialization work is only performed
+        /// once.
+        protected override async Task OnAfterRenderAsync(bool firstRender)
+        {
+            await base.OnAfterRenderAsync(firstRender);
+
+            if (firstRender)
+            {
+                this.WhenAnyValue(x => x.ViewModel.IsReadOnly).Subscribe(_=> this.StateHasChanged());

```

Comment:
create the subscription inside the ParameterSet() method is prefered since the ViewModel property is a parameter